### PR TITLE
Fix supporting gl_InstanceID.

### PR DESCRIPTION
**Pull request by xebra**
Fix again #9234

### CODE REVIEW

**Comment by xebra** [comment]
Incidentally, fixed OES_texture_npot as well.

**Comment by mrcmunir** [comment]
@xebra With mesa3d not fixed yet disabled by the check bool instanceRendering = gl_extensions.EXT_gpu_shader4;

UI/GameSettingsScreen.cpp

Also I detected if Enabled before these option always enabled  but can't disable it

captura de 2017-02-01 13-59-16


captura de 2017-02-01 13-59-39

**Comment by xebra** [comment]
@mrcmunir Ah, forgot to fix it, really thanks!

**Comment by xebra** [comment]
I modified gstate_c.supports() because it can write more simply.
If it has any issues, I'll revert it.

Allow multi flags
`if (!gstate_c.Supports(GPU_SUPPORTS_INSTANCE_RENDERING | GPU_SUPPORTS_VERTEX_TEXTURE_FETCH | GPU_SUPPORTS_TEXTURE_FLOAT))`

Allow only single flag
`if (!gstate_c.Supports(GPU_SUPPORTS_INSTANCE_RENDERING) && !gstate_c.Supports(GPU_SUPPORTS_VERTEX_TEXTURE_FETCH) && !gstate_c.Supports(GPU_SUPPORTS_TEXTURE_FLOAT))`

**Comment by mrcmunir** [comment]
Working fine with mesa3d for me no problem here :+1:

**Comment by xebra** [comment]
Thanks! :)

File: GPU/GPUState.h [hunk 1]
```diff
@@ -485,7 +485,7 @@ struct KnownVertexBounds {
 };
 
 struct GPUStateCache {
-	bool Supports(u32 flag) { return (featureFlags & flag) != 0; }
+	bool Supports(u32 flags) { return (featureFlags & flags) == flags; } // Allow multi flags
```
Comment (hrydgard) [inline GPU/GPUState.h]: good one!

Comment (xebra) [inline GPU/GPUState.h]: Thanks!

**Pull request merged**